### PR TITLE
fix route, create single typeahead component

### PR DESCRIPTION
**Pull request by drcoolsanjeev**
### Type of change

- Refactoring

### Description

- create single typeAhead component to use in all places
- use some interfaces to avoid any
- structuring code in folder format
- fix child route
- use TablePagination component instead of renderPagination

### CODE REVIEW

File: console/console-init/ui/src/components/TypeAheadSelect/TypeAheadSelect.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  SelectOption,
+  SelectOptionObject
+} from "@patternfly/react-core";
+import {
+  TypeAheadMessage,
+  TYPEAHEAD_REQUIRED_LENGTH
+} from "constants/constants";
+import { ISelectOption } from "utils";
+
+interface ITypeAheadSelectProps {
```
Comment (ajaypratap003) [inline console/console-init/ui/src/components/TypeAheadSelect/TypeAheadSelect.tsx]: an you try to use SelectProps to get all the properties, We should avoid to create unnecessary properties those already are available in importing component interface. here you can extend SelectProps and avoid the properties declare those are already threre.
interface ITypeAheadSelectProps extends SelectProps {
//declare properties which are not available in SelectProps
}

we will follow this practice for all the components.

File: console/console-init/ui/src/components/TypeAheadSelect/TypeAheadSelect.tsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+          disabled={true}
+        />
+      ) : (
+        <SelectOption
+          key={"no-results-found"}
+          value={TypeAheadMessage.NO_RESULT_FOUND}
+          disabled={true}
+        />
+      )}
+      {/* {} */}
```
Comment (ajaypratap003) [inline console/console-init/ui/src/components/TypeAheadSelect/TypeAheadSelect.tsx]: is there any significance of this commented line?

File: console/console-init/ui/src/modules/address-detail/containers/AddressLinks/AddressLinksPage.tsx [hunk 3]
```diff
@@ -118,7 +78,12 @@ const AddressLinksPage: React.FunctionComponent<IAddressLinksListPageProps> = ({
             />
           </GridItem>
           <GridItem span={5}>
-            {addresLinksTotal > 0 && renderPagination(page, perPage)}
+            <TablePagination
```
Comment (anukritijha) [inline console/console-init/ui/src/modules/address-detail/containers/AddressLinks/AddressLinksPage.tsx]: The same component is being used with different values, better way would be to create a method like renderPagination like we had earlier and pass different arguments to it in both places.

File: console/console-init/ui/src/modules/address/AddressPage.tsx [hunk 4]
```diff
@@ -342,7 +302,12 @@ export default function AddressPage() {
           />
         </GridItem>
         <GridItem span={5}>
-          {totalAddresses > 0 && renderPagination(page, perPage)}
+          <TablePagination
```
Comment (anukritijha) [inline console/console-init/ui/src/modules/address/AddressPage.tsx]: Similar to my previous comment, please encapsulate this in a method.

File: console/console-init/ui/src/components/TypeAheadSelect/TypeAheadSelect.tsx (hunk 2 above)
Comment (drcoolsanjeev) [reply console/console-init/ui/src/components/TypeAheadSelect/TypeAheadSelect.tsx]: Yes, i remove this then there will be a TS lint error. I don't know why. But some unstrange things happening

**Pull request merged**